web/gridt/tests/integration/resources/test_user.py

```diff
@@ -83,4 +82,77 @@ def test_change_password_with_correct_password(self):
 
             self.assertIn("message", resp.get_json())
             self.assertEqual(resp.status_code, 200)
-            self.assertTrue(user.verify_password("somethingyoullneverguess"))
\ No newline at end of file
+            self.assertTrue(user.verify_password("somethingyoullneverguess"))
+
+    def test_change_email_incomplete(self):
+        with self.app_context():
+            user = self.create_user()
+
+            resp = self.request_as_user(
+                self.users[0], "POST", "/user/change_email", json={}
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 400)
+
+            resp = self.request_as_user(
+                self.users[0],
+                "POST",
+                "/user/change_email",
+                json={"password": self.users[0]["password"]},
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 400)
+
+    def test_change_email_wrong_password(self):
```

Comment:
This test could then be simplified in the schema tests and only required test would be to confirm that an invalid schema would result in a 400 error.

Reply:
Then I need to set up an app context in test_schemas. Is that acceptable, or am I seeing problems where there are none?